### PR TITLE
AZ-288: Pass flags to node

### PR DESCRIPTION
**Pull request by pmikolajczyk41**

### CODE REVIEW

File: utils.py [hunk 1]
```diff
@@ -441,3 +442,8 @@ def fab_cmd():
     fabfile_path = os.environ.get('FABFILE_PATH')
     fabfile_path_flag = f'-r {fabfile_path}' if fabfile_path else ''
     return f'fab -i key_pairs/aleph.pem {fabfile_path_flag}'
+
+
+def save_node_flags(flags):
+    with open(join(os.getcwd(), 'node_flags.json'), 'w') as f:
```
Comment (kostekIV) [inline utils.py:448]: isnt it the same as `open('node_flags.json', 'w')`?

File: shell.py [hunk 2]
```diff
@@ -486,27 +486,28 @@ def setup_infrastructure(n_parties, chain='dev', regions=use_regions(), instance
 
 
 def setup_nodes(n_parties, chain='dev', regions=use_regions(), instance_type='t2.micro', volume_size=8, tag='dev',
-                **chain_flags):
+                node_flags=None, chain_flags=None):
```
Comment (kostekIV) [inline shell.py:489]: instead of None why not just dict()

File: shell.py [hunk 3]
```diff
@@ -486,27 +486,28 @@ def setup_infrastructure(n_parties, chain='dev', regions=use_regions(), instance
...
     color_print('send the binary')
     run_task('send-binary', regions, parallel, tag)
 
+    save_node_flags(node_flags or dict())
     run_task('create-dispatch-cmd', regions, parallel, tag, pids)
 
 
 def setup_benchmark(n_parties, chain='dev', regions=use_regions(), instance_type='t2.micro', volume_size=8, tag='dev',
-                    **chain_flags):
+                    node_flags=None, chain_flags=None):
```
Comment (kostekIV) [inline shell.py:506]: same here

File: shell.py (hunk 2 above)
Comment (pmikolajczyk41) [reply shell.py:489]: mutable default argument is not a good idea: https://docs.python-guide.org/writing/gotchas/

File: utils.py (hunk 1 above)
Comment (pmikolajczyk41) [reply utils.py:448]: ok, you're right: https://docs.python.org/3/library/functions.html#open

Comment (pmikolajczyk41) [reply utils.py]: changed

**Review by kostekIV — APPROVED** [review]

File: shell.py (hunk 2 above)
Comment (h4nsu) [reply shell.py:489]: I think the "pythonic" way to have mutable default arguments is to do:
```
def func(....., dict_arg=None);
     dict_arg = dict_arg or dict();
```
i.e. to do what you did, but once and on top of the function body. I don't think such a change is needed here, just dropping my 3 grosze ;)

**Review by h4nsu — APPROVED** [review]

**Pull request merged**